TypeCobol/Compiler/Sql/Model/SqlExpression.cs

```diff
@@ -39,31 +43,45 @@ public SqlConstant(Token literal)
      *                     | BinaryStringLiteral (sql-specific)
      *                     | GraphicStringLiteral (sql-specific)
      */
-    public Token Literal { get; }
-    protected override bool VisitSqlObject(ISqlVisitor visitor)
-    {
-        throw new System.NotImplementedException();
-    }
-}
 
-public enum DatetimeConstantKind
-{
-    Date,
-    Time,
-    Timestamp
-}
+        public Token Literal { get; }
 
-public class DatetimeConstant : SqlConstant
-{
-    public DatetimeConstant(Token literal,DatetimeConstantKind kind, Token tokenKind) : base(literal)
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return visitor.Visit(this);
+        }
+    }
+
+    public enum DatetimeConstantKind
     {
-        Kind = kind;
-        TokenKind = tokenKind;
+        Date,
+        Time,
+        Timestamp
     }
 
-    public override SqlConstantType Type => SqlConstantType.Datetime;
+    public class DatetimeConstant : SqlConstant
+    {
+        public DatetimeConstant(Token literal, SqlConstantType type, Token tokenKind) : base(literal,type)
+        {
+            switch (tokenKind.TokenType)
+            {
+                case (TokenType.SQL_CURRENT_DATE):
+                    Kind = DatetimeConstantKind.Date;
+                    break;
+                case (TokenType.SQL_CURRENT_TIME):
+                    Kind = DatetimeConstantKind.Time;
+                    break;
+                case (TokenType.SQL_CURRENT_TIMESTAMP):
+                    Kind = DatetimeConstantKind.Timestamp;
+                    break;
+            }
```

Comment:
The token types used here are wrong, we actually don't have token types for date, time and timestamp which are context sensitive keywords.

So I think we should have a constructor like this:

```csharp
public DatetimeConstant(Token literal, DatetimeConstantKind kind, Token tokenKind) : base(literal)
{
    [...]
}
```

The builder already test for the nature of the keyword so it knows which kind to pass.